Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -91,34 +91,18 @@ static std::optional<Length> lengthForInset(std::variant<RefPtr<CSSNumericValue>
     return { };
 }
 
-static Length lengthForInsetCSSValue(RefPtr<CSSPrimitiveValue> value)
-{
-    // TODO: Figure out how to make this work when provided calc value
-    if (!value || value->isCalculated())
-        return { };
-    if (value->valueID() == CSSValueAuto)
-        return { };
-    if (value->isPercentage())
-        return Length(value->resolveAsPercentageNoConversionDataRequired(), LengthType::Percent);
-    if (value->isPx())
-        return Length(value->resolveAsLengthNoConversionDataRequired(), LengthType::Fixed);
-
-    ASSERT_NOT_REACHED();
-    return { };
-}
-
-static ViewTimelineInsets insetsFromOptions(const std::variant<String, Vector<std::variant<RefPtr<CSSNumericValue>, RefPtr<CSSKeywordValue>>>> inset, CSSParserContext context)
+static ViewTimelineInsets insetsFromOptions(const std::variant<String, Vector<std::variant<RefPtr<CSSNumericValue>, RefPtr<CSSKeywordValue>>>> inset, RefPtr<Element> element)
 {
     if (auto* insetString = std::get_if<String>(&inset)) {
         if (insetString->isEmpty())
             return { };
         CSSTokenizer tokenizer(*insetString);
         auto tokenRange = tokenizer.tokenRange();
         tokenRange.consumeWhitespace();
-        auto consumedInset = CSSPropertyParserHelpers::consumeViewTimelineInsetListItem(tokenRange, context);
+        auto consumedInset = CSSPropertyParserHelpers::consumeViewTimelineInsetListItem(tokenRange, element->protectedDocument()->cssParserContext());
         if (auto insetPair = dynamicDowncast<CSSValuePair>(consumedInset))
-            return { lengthForInsetCSSValue(dynamicDowncast<CSSPrimitiveValue>(insetPair->protectedFirst())), lengthForInsetCSSValue(dynamicDowncast<CSSPrimitiveValue>(insetPair->protectedSecond())) };
-        return { lengthForInsetCSSValue(dynamicDowncast<CSSPrimitiveValue>(consumedInset)), std::nullopt };
+            return { SingleTimelineRange::lengthFoCSSValue(dynamicDowncast<CSSPrimitiveValue>(insetPair->protectedFirst()), element), SingleTimelineRange::lengthFoCSSValue(dynamicDowncast<CSSPrimitiveValue>(insetPair->protectedSecond()), element) };
+        return { SingleTimelineRange::lengthFoCSSValue(dynamicDowncast<CSSPrimitiveValue>(consumedInset), element), std::nullopt };
```

Comment:
Same here, I'd potentially try to split this into multiple lines.

Reply:
Fixed.